test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -83,31 +87,56 @@ public void EnlistOff()
             using (var conn2 = OpenConnection(ConnectionStringEnlistOff))
             {
                 Assert.That(conn1.EnlistedTransaction, Is.Null);
-                Assert.That(conn1.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test')"), Is.EqualTo(1));
-                Assert.That(conn2.ExecuteScalar("SELECT COUNT(*) FROM data"), Is.EqualTo(1));
+                Assert.That(conn1.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test')"), Is.EqualTo(1),
+                    "Unexpected insert rowcount");
+                Assert.That(conn2.ExecuteScalar("SELECT COUNT(*) FROM data"), Is.EqualTo(1),
+                    "Unexpected data count");
+            }
+
+            // Scope disposed and not completed => rollback, but no enlistment, so changes should still be there.
+            using (var conn3 = OpenConnection(ConnectionStringEnlistOff))
+            {
+                Assert.That(conn3.ExecuteScalar("SELECT COUNT(*) FROM data"), Is.EqualTo(1), "Insert unexpectedly rollback-ed");
```

Comment:
If the `Enlist=false` does not work and enlist connections, both would be in same transaction and thus should see the same data. Conn2 would then read the uncommitted data from conn1. So it would not ascertain they are not enlisted.